src/Encoder.c

```diff
@@ -3132,7 +3132,22 @@ static ZyanStatus ZydisEmitUInt(ZyanU64 data, ZyanU8 size, ZydisEncoderBuffer *b
     }
 
     // TODO: fix for big-endian systems
-    ZYAN_MEMCPY(buffer->buffer + buffer->offset, &data, size);
+    if (size == 1)
+    {
+        ZYAN_MEMCPY(buffer->buffer + buffer->offset, &data, 1);
+    }
+    else if (size == 2)
+    {
+        ZYAN_MEMCPY(buffer->buffer + buffer->offset, &data, 2);
+    }
+    else if (size == 4)
+    {
+        ZYAN_MEMCPY(buffer->buffer + buffer->offset, &data, 4);
+    }
+    else if (size == 8)
+    {
+        ZYAN_MEMCPY(buffer->buffer + buffer->offset, &data, 8);
+    }
 
```

Comment:
Might be worth to experiment with ZYAN_UNREACHABLE in an unconditional 'else' branch. Could further improve code generation in the best case and warn us about incorrect usage in the future.